examples/UI.Note-sample.xml

```diff
@@ -0,0 +1,58 @@
+<?xml version="1.0" encoding="utf-8"?>
+<edmx:Edmx Version="4.0" xmlns:edmx="http://docs.oasis-open.org/odata/ns/edmx">
+  <edmx:Reference Uri="/sap/opu/odata4/sap/ui_nte_demo_m_o4/srvd/sap/ui_nte_demo_m/0001/$metadata">
+    <edmx:Include Namespace="com.sap.gateway.srvd.ui_nte_demo_m.v0001" Alias="service" />
+  </edmx:Reference>
+  <edmx:Reference Uri="https://oasis-tcs.github.io/odata-vocabularies/vocabularies/Org.OData.Core.V1.xml">
+    <edmx:Include Namespace="Org.OData.Core.V1" Alias="Core" />
+  </edmx:Reference>
+  <edmx:Reference Uri="https://sap.github.io/odata-vocabularies/vocabularies/UI.xml">
+    <edmx:Include Namespace="com.sap.vocabularies.UI.v1" Alias="UI" />
+  </edmx:Reference>
+  <edmx:Reference Uri="https://sap.github.io/odata-vocabularies/vocabularies/Common.xml">
+    <edmx:Include Namespace="com.sap.vocabularies.Common.v1" Alias="Common" />
+  </edmx:Reference>
+  <edmx:DataServices>
+    <Schema Namespace="UI.examples" xmlns="http://docs.oasis-open.org/odata/ns/edm" Alias="self">
+      <Annotations Target="service.C_NTE_DEMO_MType">
+        <Annotation Term="UI.Facets">
+          <Collection>
+            <Record Type="UI.ReferenceFacet">
+              <PropertyValue Property="Target" AnnotationPath="_NoteBasic/@UI.Note" />
+            </Record>
+          </Collection>
+        </Annotation>
+      </Annotations>
+      <Annotations Target="service.C_NoteBasicTPType">
+        <Annotation Term="UI.Note">
+          <Record>
+            <PropertyValue Property="Title" Path="NoteBasicTitle" />
+            <PropertyValue Property="Content" Path="NoteBasicContent" />
+            <PropertyValue Property="Type" Path="NoteBasicType" />
+            <PropertyValue Property="MaximalLength" Path="_NoteBasicType/NoteBasicMaximalLength" />
+            <PropertyValue Property="MultipleNotes" Path="_NoteBasicType/NoteBasicMultipleNotes" />
+          </Record>
+        </Annotation>
+        <Annotation Term="Common.SAPObjectNodeType">
+          <Record>
+            <PropertyValue Property="Name" Path="_NoteBasicType/NoteBasicObjectNodeType" />
+          </Record>
+        </Annotation>
+        <Annotation Term="Common.CreatedBy" Path="CreatedByUser" />
+        <Annotation Term="Common.CreatedAt" Path="CreationDateTime" />
+        <Annotation Term="Common.ChangedBy" Path="LastChangedByUser" />
+        <Annotation Term="Common.ChangedAt" Path="LastChangeDateTime" />
+      </Annotations>
+      <Annotations Target="service.C_NoteBasicTPType/NoteBasicTitle">
+        <Annotation Term="UI.Hidden" Path="_NoteBasicType/NoteBasicHideTitle" />
+      </Annotations>
+      <Annotations Target="service.C_NoteBasicTPType/NoteBasicContent">
+        <Annotation Term="Core.MediaType" Path="NoteBasicMIMEType" />
```

Comment:
```suggestion
        <Annotation Term="Core.MediaType" Path="NoteBasicMIMEType" />
        <Annotation Term="Validation.Pattern">
          <Apply Function="odata.concat">
            <String>.{,</String>
            <Path>_NoteBasicType/NoteBasicMaximalLength</Path>
            <String>}</String>
          </Apply>
        </Annotation>
```
This would allow us to omit the `@UI.Note/MaximalLength` property.

Reply:
could you give an example how this string would look like?
Would it be something like this: `{NOTETYPE1,200},{NOTETYPE2,50}`? 
If yes I guess, it would be ok for us.

Reply:
`{"@Validation.Pattern": ".{,128}"}`

Reply:
ok, now I got it. This validation pattern is assigned to the NoteContent.
This means for the content I have to follow the path `_NoteBasicType/NoteBasicMaximalLength` and I get the maxLength, right?
The question is only how can I define this in CDS?

Reply:
> This means for the content I have to follow the path `_NoteBasicType/NoteBasicMaximalLength` and I get the maxLength, right?

Well, you would have to ensure that the `<Apply>` element otherwise looks like in this example. I find this tooo tricky.